.github/workflows/deploy.yml

```
@@ -0,0 +1,40 @@
+name: Deploy to VSCode marketplace
+
+on:
+  release:
+    types: [published]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  deploy:
+    runs-on: windows-latest
+    timeout-minutes: 5
+    
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2.1.0
+      with:
+        submodules: true
+    
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v1.4.2
+      with:
+        node-version: '12.x'
+    
+    - name: Check release version matches package.json
+      uses: ./.github/actions/check-package-version
+
+    - name: Install dependencies
+      run: npm install
+
+    - name: Publish to VSCode Marketplace
+      env:
+        PAT: ${{ secrets.MARKETPLACE_PAT }}
+      run: npx vsce publish -p "$PAT"
```

Can we avoid using an environment variable to reduce the exposure of the PAT?

```suggestion
      run: npx vsce publish -p "${{ secrets.MARKETPLACE_PAT }}"
```

---

I was following the suggested way on https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow

---

Fair enough, their justification is that you can scrape arguments from `ps`.